src/backend/tests/test_form_filling.py

```diff
@@ -31,3 +35,351 @@ def test_normal_conviction_uses_multnomah_conviction_form():
+            ],
+        )
+
+    def test_pdf_boolean_has_class_c_felony(self, pdf: PDF):
+        assert_pdf_boolean_field(
+            pdf,
+            "has_class_c_felony",
+            [
+                "(Felony  Class C and)",
+                "(5 years have passed since the later of the convictionjudgment or release date and)",
+                "(I have not been convicted of any other offense or found guilty except for insanity in_2)",
+            ],
+        )
+
+    def test_pdf_boolean_has_class_a_misdemeanor(self, pdf: PDF):
+        assert_pdf_boolean_field(
+            pdf,
+            "has_class_a_misdemeanor",
+            [
+                "(Misdemeanor  Class A and)",
+                "(3 years have passed since the later of the convictionjudgment or release date and)",
+                "(I have not been convicted of any other offense or found guilty except for insanity in_3)",
+            ],
+        )
+
+    def test_pdf_boolean_has_class_bc_misdemeanor(self, pdf: PDF):
+        assert_pdf_boolean_field(
+            pdf,
+            "has_class_bc_misdemeanor",
+            [
+                "(Misdemeanor  Class B or C and)",
+                "(1 year has passed since the later of the convictionfindingjudgment or release)",
+                "(I have not been convicted of any other offense or found guilty except for insanity)",
+            ],
+        )
+
+    def test_pdf_boolean_has_violation_or_contempt_of_court(self, pdf: PDF):
+        assert_pdf_boolean_field(
+            pdf,
+            "has_violation_or_contempt_of_court",
+            [
+                "(Violation or Contempt of Court and)",
+                "(1 year has passed since the later of the convictionfindingjudgment or release_2)",
+                "(I have not been convicted of any other offense or found guilty except for insanity_2)",
+            ],
+        )
+
+    def test_pdf_boolean_has_probation_revoked(self, pdf: PDF):
+        assert_pdf_boolean_field(
+            pdf,
+            "has_probation_revoked",
+            [
+                "(I was sentenced to probation in this case and)",
+                "(My probation WAS revoked and 3 years have passed since the date of revocation)",
+            ],
+        )
+
+    # TODO some fields seem to need a different calculation
+    # def test_font_is_decreased_as_needed(self, pdf: PDF):
+    #     form_data = {"date_of_birth": "123456789a1234567"}
+    #     pdf.update_annotations(form_data, {"assert_blank_pdf": True});
+    #     font_string = [annotation.DA for annotation in pdf.annotations if annotation.T == '(DOB)'][0]
+
+    #     pdf.write('foo_test')
+    #     assert font_string == "(/TimesNewRoman 6 Tf 0 g)"
+    #     assert len(pdf.warnings) == 1
+    #     assert pdf.warnings[0] == "The font size of \"123456789a123456\" was shrunk to fit the bounding box of \"DOB\". An addendum might be required if it still doesn't fit."
```

Comment:
I prefer to not have commented code especially if the git history can preserve it. Was this intentionally left behind?

Reply:
Sounds good!